packages/cozy-client/src/ClientErrors.jsx

```diff
@@ -0,0 +1,118 @@
+import { pull } from 'lodash'
+
+/**
+ * Rendering functions for client fetch errors by status code
+ *
+ * @private
+ * @type {object}
+ */
+const byHttpStatus = {
+  413: QuotaError
+}
+
+/**
+ * Display for a quota error from the client
+ *
+ * @see QuotaAlert
+ * @private
+ * @param {Function} dismiss - remove the error from the stack to display
+ * @returns {Function} React rendering
+ */
+function QuotaError({ dismiss }) {
+  return <QuotaAlert onClose={dismiss} />
+}
+
+/**
+ * Returns the handler for an error
+ *
+ * @param {Error} error -
+ * @returns {Function|null} React Component
+ */
+function getErrorComponent(error) {
+  if (error instanceof Response) {
+    return byHttpStatus[error.status] || null
+  }
+  return null
+}
+
+/**
+ * Renders a stack of errors
+ *
+ * @private
+ * @see ClientErrors
+ * @param {Error[]} errorStack - array of errors/exceptions
+ * @param {Function} setErrorStack - mutates the array of errors
+ * @returns {Function} React rendering
+ */
+function renderErrors(errorStack, setErrorStack) {
+  return errorStack.map((error, key) => {
+    const Component = getErrorComponent(error)
+    if (Component) {
+      const dismiss = () => setErrorStack(stack => pull(stack, error))
+      return <Component key={key} error={error} dismiss={dismiss} />
+    } else {
+      return null
+    }
+  })
+}
+
+/**
+ * Manage the client errors and exceptions, display them if necessary
+ *
+ * Only Quota Errors are managed for now.
+ *
+ * @param {object} props -
+ * @param {boolean} props.handleExceptions - should cozy-client directly handle errors before forwarding them to the caller?
+ * @param {Function} props.children - React rendering children
+ * @returns {Function} React rendering
+ */
+export function ClientErrors(props) {
+  const { handleExceptions = true, children } = props
+
+  const client = useClient()
+  const [errorStack, setErrorStack] = useState([])
+
+  /**
+   * Handle client errors, add them to the error stack
+   *
+   * @param {Error} error -
+   * @returns {boolean}
+   */
+  const handleError = useCallback(
+    error => {
+      // `isErrorManaged` is there to avoid the same error to be added twice
+      // once the error has been added once, the `isErrorManaged`is set to true
+      // and any future push is ignored
+      if (error.isErrorManaged) return false
+
+      if (error instanceof Response) {
+        const isManaged = !!getErrorComponent(error)
+        if (isManaged) {
+          error.isErrorManaged = true
+          setErrorStack(stack => stack.concat(error))
+          return true
+        }
+      }
+      error.isErrorManaged = false
+      return false
+    },
+    [setErrorStack]
+  )
+
+  useEffect(() => {
+    if (handleExceptions) {
+      client.on('error', handleError)
+      return () => client.removeListener('error', handleError)
+    }
+  }, [client, handleError, handleExceptions])
+
+  return (
+    <>
+      {children}
+      {renderErrors(errorStack, setErrorStack)}
```

Comment:
I am wondering if we could change the component not be a Wrapper component = that it should not render its children, I am a bit uneasy with having a deep React tree, and am not sure it is necessary here since we do not use a Provider.